### DISCUSSION TITLE
How to use Pybricks after installation

### DISCUSSION THREAD

**Discussion by David-Eriksson** [pybricks/support#1416] [Q&A]
Hi,

I tried to install pybricks firmware for 51515 hub on samsung table A7, chrome browser. Pushing Bluetooth, colors flashing, connecting USB, release Bluetooth, and press install...

During installation, the browser window became blank white.

Now when I turn on the hub it shows a 3x3 square and the main button is pulsating blue (two short blinks and a break, and again two short blinks, etc.). The hub is somehow detected by the operating system, but one cannot connect to it.

What can I do?

Best regards,
David

**Comment by laurensvalk** [comment]
> Now when I turn on the hub it shows a 3x3 square and the main button is pulsating blue 

This means installation was successful.

Instead of connecting via your operating system, use the Bluetooth button in the Pybricks app:

![image](https://github.com/pybricks/support/assets/12326241/b8c2fdcb-819e-4aee-9ce5-2e2a43ecb8e9)

And since your installation was successful, you could always go back to the original firmware, since that works exactly the same way:

![image](https://github.com/pybricks/support/assets/12326241/4e45cdac-f862-4b4a-9bb8-98da6f1c51c2)

**Reply by kingananas20** [reply]
How can I connect to it without the APP? I want to run it with pybricksdev run command. Is that possible?

**Comment by David-Eriksson** [comment]
Thanks so much for the quick response! That is obviously correct, my
stupidity.

My next stupidity is that I got a license for the text block programming
before testing if it works... When I go to the beta I can see the python
code but when I open a text block I see nothing but a turning/loading wheel
in the coding windows... Hmm? May I ask for which hardware /operating
system it works?

Best,
David

laurensvalk ***@***.***> schrieb am So., 4. Feb. 2024, 13:05:

**Reply by laurensvalk** [reply]
If you've already managed to come this far, the odds are everything else will work too :smile:

The only thing that is operating system dependent is connectivity (USB installation and Bluetooth for coding), but it sounds like you've got that working. 

If you've just opened the app (or its beta variant) it may take a little while for the block editor to load. The next time is should be nearly instant, since it's saved for offline use.

If the issue persists, can you share a screenshot of what you are seeing?

**Comment by David-Eriksson** [comment]
That is comforting.

Attached is a screenshot when I have selected a block py file.

After 2 minutes the browser gets blank white. Please find the other
attached file. Hmmm.

I will sleep on it and maybe it goes better tomorrow.

Best,
David




laurensvalk ***@***.***> schrieb am So., 4. Feb. 2024, 17:05:

**Reply by dlech** [reply]
GitHub discards attachments from emails. You can upload files using the web interface instead.

**Comment by David-Eriksson** [comment]
Ok. Here they are. Screenshots are from my Samsung SmartPhone, but it also happens on Samsung Tablet A7.
![After2Minutes](https://github.com/pybricks/support/assets/50678197/030b1ba8-83df-4986-afa6-dd6e31410666)
![First2Minutes](https://github.com/pybricks/support/assets/50678197/0c034378-167c-4fb5-86ca-36eb6d9a2e5d)

**Reply by laurensvalk** [reply]
This is very unusual, thanks for sharing this!

Does this also happen on https://code.pybricks.com/ ?

**Reply by laurensvalk** [reply]
After trying that, it may be worth resetting the Pybricks app, using the instructions below. Thanks in advance for giving this a try.

Note:
- this will delete your Pybricks programs if you've already made any.
- your license will be deactivated. You can just activate the same code again.

Open your browser settings using the **⋮** button on the top right:

![image](https://github.com/pybricks/support/assets/12326241/ee17a5f4-e7ce-424f-833a-abc8bba73876)

Then go to site settings:

![image](https://github.com/pybricks/support/assets/12326241/101423f1-436e-4caa-8b25-c7aa0dd65764)

Search for Pybricks on the list of sites. Then choose the trashcan to erase all Pybricks data.

![image](https://github.com/pybricks/support/assets/12326241/dc303e62-3afd-4ffe-bc3a-de4e684c6b45)

Then go to https://beta.pybricks.com/ again to see if it made a difference. 

On the other hand, since this happens on both of your devices, maybe this won't help. Do you have a PC to try it on, to make sure the web connection isn't the issue?

**Comment by David-Eriksson** [comment]
On the PC it works perfectly. On the Android devices, I also have the
problem that many of the icons (the Bluetooth icons, for example) are not
displayed. Similarly, the option of selecting code or block-based comes
with nice illustrations on the PC, but on the Android, there are no
illustrations but only two radio buttons. The same thing for choosing which
type of hub for the firmware update. I have tested it with Samsung browser
and chrome, same thing.


Am Mo., 5. Feb. 2024 um 10:04 Uhr schrieb laurensvalk <
***@***.***>:

**Reply by laurensvalk** [reply]
Thank you, this is helpful. Then maybe it is not related to block coding per-se.

It sounds like the app is not loading entirely. That results in missing images, and missing portions of the code that are loaded in the background. When it fails to "lazy-load" the background code, it would explain the white screen.

So in this case it could be worth testing the reset steps outlined above.

Also, which version of Chrome and Android do you have on your devices? To find out, again to go the ⋮and scroll down to "About Chrome".

For example I see this:

![Screenshot_20240205-150501~2.png](https://github.com/pybricks/support/assets/12326241/8b570741-db70-4d0e-9e91-1eba59dd4294)

![Screenshot_20240205-150534~2.png](https://github.com/pybricks/support/assets/12326241/ccce6acf-b700-4e56-b1fe-82f3a66ec2b9)

**Comment by David-Eriksson** [comment]
I didn't know that one cannot use the wifi at the same time!? I turned off
Wifi on my smartphone and it worked. Hmm? Sorry for finding out so late!
But why is this? Is there a possibility that this will be changed in the
future? I mean if one need wifi for internet, which in turn is required for
starting the pybricks. Once started one can run it in offline mode which is
great of course.

laurensvalk ***@***.***> schrieb am Mo., 5. Feb. 2024, 15:06:

**Reply by laurensvalk** [reply]
You should be able to just keep WiFi on though. Maybe it just finished loading this time around.

Could you check the screenshots above to verify your Android version, and the app reset option? Thank you!

**Comment by David-Eriksson** [comment]
Ohh, sorry, I overlooked that request! For my smartphone:
Chrome 121.0.6167.143
Android 12; SM-M215F Build/SP1A.210812.016

I will try to check my tablet today evening.

laurensvalk ***@***.***> schrieb am Di., 6. Feb. 2024, 10:47:

**Reply by laurensvalk** [reply]
Thank you, that looks recent enough, so that part of it looks fine.

If you experience the load issue on your tablet still, it would be great if you want to try the reset steps on your tablet. Thanks!

**Comment by David-Eriksson** [comment]
Dear Laurensvalk,

I turned off the web language translation, then, suddenly the
bluetooth-connected button icon appeared. Then I updated chrome. Then after
a couple of trials everything was loaded. Now it works perfectly!

This is an amazing piece of software (my son is only missing the variable
lists and functions, but I guess that is coming :-)) with a clear focus on
education and inspiring children. I seriously think this is one of the few
hopes to get children away from facebook and tiktok. The problem, of
course, is the expensive hardware and maybe the lack of interest from the
parents. On this note, I think there is no excuse for schools not having at
least one of those hubs. Then, children can see that the fascinating
smartphone can be used for things other than social media and play games.
Indeed pybricks is like a game with the important distinction that it has a
connection to the physical world with the robotics aspect.

Overall, children should get introduced to pybricks/smartphone/lego a
little earlier than social media so that the magic of the latter is reduced.

Sorry, you may remove this from the github discussion :-).

Best regards,
David

Am Di., 6. Feb. 2024 um 10:59 Uhr schrieb laurensvalk <
***@***.***>:

**Comment by David-Eriksson** [comment]
For completeness, I forgot to tell you that I deleted the app specific data
in the browser, but it did not help. The only thing that helped was to try
to make the loading "easier" by for example turning off Wifi, or
translation, or updating browser as you suggested.

laurensvalk ***@***.***> schrieb am Di., 6. Feb. 2024, 10:59:
